System info when clicking the Bonsai Info

falken10vdl opened the pull request:
As described here https://community.osarch.org/discussion/2897/add-system-info-to-clipboard-for-easy-referencing#latest call bpy.ops.bim.copy_debug_information() operator for easy review the system information.
Thanks!

maxfb87 committed:
Purge unecessary code

Andrej730 committed:
typing

falken10vdl committed:
add operator to copy system info to clipboard

falken10vdl committed:
add operator to copy system info to clipboard

falken10vdl committed:
simplified as per community feedback

falken10vdl committed:
use invoke_popup instead of invoke_props_dialog

Andrej730 committed:
Fix resaving camera to IFC on every drawing activation (32899eb)

`camera_props.update_representation` basically always returned True leading to `bim.update_representation` call on camera on every drawing activation. It was always True because `create_camera` wasn't setting `representation` and some other related camera props.

Changes:
- create_camera now setting representation
- ortho_scale, rather_x and raster_y now explicitly is tied to width, height and drawing scale props, not on depsgraph update. Depsgraph update wasn't allowing to set these props in time for create_camera to record representation correctly
- depsgraph update now only in charge of ortho_scale (just to be sure user won't touch it by hand) and scene.render resolution

Andrej730 committed:
create_camera - small optimization with early return

Andrej730 committed:
create_camera - small optimization with util.shape

Andrej730 committed:
IFC Mesh Separate - add to object mode menu too for discoverability + support multiple selected elements

Short demo - https://imgur.com/a/PIuJ9Rq

Andrej730 committed:
core tests - more readable errors

E.g. it shows now:
AttributeError: Interface 'bonsai.core.tool.Drawing' has no attribute 'weird_method'.

Instead of:
Prophecy <class 'abc.Drawing'> has no attribute weird_method

Andrej730 committed:
typing

maxfb87 committed:
Fix bug in reload link ifc where in Windows gives an error if relative path is used

Andrej730 committed:
core tests - very readable error for unpredicted return values

Previously:
E       TypeError: Object of type Prophecy is not JSON serializable

Now:
E           TypeError: Failed to serialize call: '{'name': 'duplicate_ifc_objects', 'args': ([<Prophecy for 'bonsai.core.tool.Ifc'>],), 'kwargs': {}}'.
E           Possibly due to unpredicted return value for some call.
E           See the list of the recent calls without return values:
E           - {'name': 'get_object', 'args': ('annotation',), 'kwargs': {}}
E           - {'name': 'run', 'args': ('group.assign_group',), 'kwargs': {'group': 'new_group', 'products': ['new_drawing']}}
E           - {'name': 'run', 'args': ('group.edit_group',), 'kwargs': {'group': 'new_group', 'attributes': {'Name': 'unique_name', 'ObjectType': 'DRAWING'}}}
E           - {'name': 'run', 'args': ('group.unassign_group',), 'kwargs': {'group': 'group', 'products': ['new_drawing']}}

Andrej730 committed:
Fix core test for drawing duplication (0b1c049)

Andrej730 committed:
core.duplicate_drawing - remove debug print statements and unused code

Andrej730 committed:
IFC Mesh Separate - reload representation just once if multiple items got separated

Andrej730 committed:
IFC Mesh Separate - fix bug when separating an item wouldn't change separated item

Mentioned by Ryan in #6285

Andrej730 committed:
entity_instance.__getattr__ - document some attributes conversion to Python

Andrej730 committed:
IFC Mesh Separate - fix bug when new separated mesh wasn't relinked to a new ifc item

Andrej730 committed:
bim.import_representation_items - add description

Andrej730 committed:
IFC Mesh Separate - fix not updated items decorator after some item was separated

Noticed fixing bug from #6285

Andrej730 committed:
IFC Mesh Separate - fix bug in 52af302bef separating multiple elements

Forgot to clear selection after processing each object, it could have led to incorrect results since mesh.separate is applied to the selection later.

Andrej730 committed:
Revert "Bump IOS build in Bonsai" #6589

This reverts commit 50e7728f95e8fa36193303c51c3d58725be33fa9.

c4rlosdias committed:
clear list of entities failed when execute Tester

theoryshaw committed:
Fix #6601 - If you duplicate something it makes the duplicate the active object.
also bulk deselect all objects_to_remove to speed things up a little.

theoryshaw committed:
small tweak to previous commit

theoryshaw committed:
whoops

Andrej730 committed:
IFC Join - add poll check similar to object.join

Andrej730 committed:
typing

Andrej730 committed:
IFC Join - add description

Andrej730 committed:
IFC Join - prevent possible ghost ifc objects after join

Skip ifc objects of different type as they will be ignored by object.join either way and removing them produced ghost ifc objects.

Andrej730 committed:
IFC Join - prevent joining rep items into non items

Because it was removing representation item object leaving Bonsai in confusing state, which may lead to errors later

Andrej730 committed:
IFC Join - fix error joining representation items (6ce4b5d)

It was trying to reaccess already removed ifc_definition_id leading to an error. Now we update id with a new item.

Andrej730 committed:
geometry.name_item_object

Andrej730 committed:
reuse ifc.link for meshes

Andrej730 committed:
edit_meshlike_item - fix missing item object name update

Andrej730 committed:
IFC Join - support joining non-ifc Blender objects into representation items

Short demo - https://imgur.com/a/o9MJiIr

brunopostle committed:
Linux escape double-quotes in filenames #6630

Special chars ('#{)$ were already escaped, but " wasn't

aothms committed:
Mark settings as internal #6623

aothms committed:
Don't set xstep.cascade.unit as it appears to result in scaling during export #6623

Andrej730 committed:
get_uri, resolve_uri - refactor #6605

1) Use pathlib for consistency
2) Ensure consistent output - always use posix format and resolve paths (but keep the symlinks)

Andrej730 committed:
Fix issues printing drawing with projects linked by relative path

The fix in f9bf889 wasn't related to this problem and was targetting another bug.

Andrej730 committed:
Fix missed cases of loading unloaded linked files in 9316e0b (#6403)

Andrej730 committed:
typing

Andrej730 committed:
Revert f9bf889 #6605

As this fix is no longer needed and covered by more general resolve_uri refactor

Andrej730 committed:
import_ifc - skip setup_arrays for annotations #6636

Andrej730 committed:
Setup array constraints for annotations that use paramatric arrays

Previously you was able to add parametric array to annotations but constraint would be missing on file reload.

Andrej730 committed:
bim.add_array - guard from unexpected element types

So we can gradually expand it with the things we do know to support.

Andrej730 committed:
mesh.add_grid - display props in Blender project units

Example - https://i.imgur.com/koLiH7U.png

Andrej730 committed:
Use actual text position instead of local pos rendering drawings text decorations

E.g. it's important if text is part of parametric array and it's local position may be constant, but it's constrainet by parent's position.

aothms committed:
One more self-intersection check in cgal kernel #6589

aothms committed:
todo notes #6589

aothms committed:
partial_success_is_success #6589

Andrej730 committed:
Operator to run git diff on current ifc file

Short demo - https://imgur.com/a/TvnztNy

Andrej730 committed:
create_axis_curve - fix typo

Andrej730 committed:
ci-black-formatting - check syntax errors explicitly

FabioPiccinini committed:
Use tool.Blender.get_view3d_area() to allow the use of separate windows (#6617)

Andrej730 commented:
Looks good in general, there are only couple small things to fix.
And can you please run black formatter on this (just run `black .` in the repository root directory)? 
ci-black-formatting is kind of confusing now and never reports if code is not formatted, we'll need to fix ir.

falken10vdl commented:
Sure. Let me do it and report back.
Thanks!

aothms committed:
partial_success_is_success #6589

falken10vdl commented:
<img width="658" alt="image" src="https://github.com/user-attachments/assets/11b1019a-a36f-4029-ab2b-b49dd1d5d2c1" />

(the search/operator.py has nothing to do I have undone the changes)
What do I do now? :) I send another PR? Is it possible to edit the current one? Sorry I am very new to handling code in github.
In general before submitting PRs I must run black . in the IFCopenshell root directory, correct?
Thanks!

Andrej730 commented:
> I must run black . in the IFCopenshell root directory, correct?

yes


> What do I do now? :) I send another PR? Is it possible to edit the current one? Sorry I am very new to handling code in github.

Just commit the changes to the branch you were working on (https://github.com/falken10vdl/IfcOpenShell/tree/system_info) and push them.

falken10vdl committed:
black formatting applied to 3 files.

Andrej730 committed:
typing

Andrej730 committed:
IfcGrid - add placement by default to avoid invalid IFC

Andrej730 committed:
Fix invalid IfcGrid cartesian points (774a770)

They were saved as 4d coordinates, not 2d.

Andrej730 committed:
black .

aothms committed:
Fix faceset helper on advanced brep oriented edges #6604

aothms committed:
Defensiveness against empty faces

aothms committed:
Bounds checking in variant array

Andrej730 committed:
Fix typo

Andrej730 committed:
More consistent Windows Bonsai .ifc file association

Rely on `winreg` instead of running powershell scripts with cmd commands + .reg file + weird workflow with the need to retype cmd command. Also use HKCU instead of HKCR because it doesn't require admin rights.

If someone hearing about this feature for the first time, here's a short demo - https://imgur.com/a/dfG44pY (icon appears too but after explorer restart).

Andrej730 committed:
Windows - fix opening paths with special characters #6630

Andrej730 committed:
Fix bug creating a drawing if linked model schema doesn't match the main model schema

Co-Authored-By: Thomas Krijnen <1096535+aothms@users.noreply.github.com>

Andrej730 committed:
Reenable black formatting workflow

Since 3b1e893 black formatter workflow basically had no use since it wasn't reporting any issues (no one checks workflow logs unless workflow failed) and wasn't reliable for catching syntax errors either (black formatter catches syntax errors only when they get in the way of it's formatting the code, but now we delegate to python itself to catch syntax errors, see 406962287c).

And we also have some docs on how to apply black formatter - https://docs.bonsaibim.org/guides/development/code_style.html

Andrej730 committed:
new operator - bim.select_entity_by_guid

I find myself often having element guid copied (either from other viewer or from text editor or from elsewhere) and to find it in scene I needed to create a new search group, add a filter, paste guid and then search for it. So decided to add a shortcut for this that just takes guid from the clipboard and runs the search for it.

Location - https://i.imgur.com/3c5rm1N.png

Andrej730 committed:
Reuse get_last_commit_hash

Noticed that hash wasn't updating from live git repo because of this.
Also update commit data only when running the package, not extension.

Andrej730 committed:
Display actual Bonsai version for live repo (reuse last commit date)

Andrej730 committed:
color by property - add operators descriptions

Andrej730 committed:
typing

Andrej730 committed:
Fix ifc2x3 errors editing IfcSurfaceStyleShading #6624

Andrej730 committed:
Fix dangerous bug with bim.override_outliner_delete

Essentialy outliner.delete should do the same as object.delete, but in our case the main implementation was in object.delete while outliner.delete was missing very important parts - e.g. handling arrays.

So if user could have remove part of an array without errors, leaving pset data invalid. See #6590, #6547.
Found this by trying to debug issues above.

Now outliner.delete will call object.delete explicitly, so the two stay connected.

I also had an idea that may be it somehow can produce #6632/#6619, but couldn't find a way to reproduce it. So it's still uncertain.

Andrej730 committed:
overridden delete operators - add descriptions

Andrej730 commented:
Also please rebase using main branch after submitting changes to issues mentioned in review comments, so PR can be merged without conflicts.

falken10vdl committed:
update after rebase

Andrej730 commented:
Something odd happened during rebase - in theory it should have used the main branch and added your commits at the end, but it seem to recommitted all the commits in between instead.

![image](https://github.com/user-attachments/assets/ddd2e0d8-6df4-4fd5-86cc-8f11cd8f7f89)

falken10vdl commented:
Yes, that is what I was expecting as well. I was doing it from within Vscode.
Sorry for that. I might have done something wrong. 
Maybe easiest is to close this one and just go for a fresh one:  https://github.com/IfcOpenShell/IfcOpenShell/pull/6660
